### DISCUSSION TITLE
Can I use multiple imageLoaders?

### DISCUSSION THREAD

**Discussion by 629Hz** [coil-kt/coil#2737] [Q&A]
Hello

I need to implement a separate disk cache, certain images are important and shouldn't be wiped out by other caches, so I'm going to have two cache directories. I think the memory cache is fine to share.

``` kotlin
val imageLoader1 =  ImageLoader.Builder(context)
    .diskCache {
        DiskCache.Builder()
            .directory(context.cacheDir.resolve("image_cache"))
            .maxSizeBytes(DISK_CACHE_MAX_SIZE)
            .build()
    }
    .build()

val imageLoader2 =  ImageLoader.Builder(context)
    .diskCache {
        DiskCache.Builder()
            .directory(context.cacheDir.resolve("important_image_cache"))
            .maxSizeBytes(DISK_CACHE_MAX_SIZE)
            .build()
    }
    .build()
```

Can I use different image loaders with two disk cache directories?
Thanks

**Comment by colinrtwhite** [comment]
Yep! You can also use `ImageLoader.newBuilder` to share resources like the memory cache.
